### ISSUE TITLE
Dateutil version

### ISSUE THREAD

**Issue by luisalima** [facebook/ThreatExchange#134]
Hi guys,

I ran:

```sh
python2.7 setup.py install
```

Which ended successfully.

Running your example as is yields the following error in my system:

```sh
> python ./get_all_data/get_threat_indicators.py --text="facebook" --days_back=10
Traceback (most recent call last):
  File "./get_all_data/get_threat_indicators.py", line 65, in <module>
    main()
  File "./get_all_data/get_threat_indicators.py", line 33, in main
    utils.get_time_params(s.end_date, day_counter, format_)
  File "build/bdist.macosx-10.11-x86_64/egg/pytx/utils.py", line 62, in get_time_params
AttributeError: 'module' object has no attribute 'parser'
```

To fix it, I changed `utils.py` as follows:

```diff
diff --git a/pytx/pytx/utils.py b/pytx/pytx/utils.py
index a42dc50..5538f3d 100644
--- a/pytx/pytx/utils.py
+++ b/pytx/pytx/utils.py
@@ -1,4 +1,4 @@
-import dateutil
+import dateutil.parser as dateutil_parser
 import datetime


@@ -59,7 +59,7 @@ def get_time_params(end_date, day_counter, format_):
     """
     # We use dateutil.parser.parse for its robustness in accepting different
     # datetime formats
-    until_param = dateutil.parser.parse(end_date) - \
+    until_param = dateutil_parser.parse(end_date) - \
         datetime.timedelta(days=day_counter)
     until_param_string = until_param.strftime(format_)
```

And then I ran `python2.7 setup.py install` again, after which the `get_threat_indicators.py` ran successfully.

I noticed that, in `setup.py`, and also in `requirements.txt` (why the duplication?) you are specifying the required version of `dateutil` as 2.5.2, but that seems to be ignored by the system. I'm not an expert in python builds so maybe all of this is not that useful for you, but I thought I'd give you guys a heads up. Please close if irrelevant. Thanks for the very interesting work 😄

**Comment by mgoffin** [comment]
Checking this out now!

I did just try an install and saw that dateutil was being installed (in my case it already existed):

```
Searching for python-dateutil==2.5.2
Best match: python-dateutil 2.5.2
Processing python_dateutil-2.5.2-py2.7.egg
Removing python-dateutil 2.4.2 from easy-install.pth file
python-dateutil 2.5.2 is already the active version in easy-install.pth
```

When you say it's being ignored by the system what are you seeing? The duplication in the `requirements.txt` and `requirements-dev.txt` file are for testing (builds in Travis) and installation flexibility. For example, someone can run `pip -r requirements.txt` if they wish for testing dependency installation outside of what `setup.py` is doing.

**Comment by mgoffin** [comment]
I've added PR #135. I wasn't able to replicate the error you got using the command you provided, but I was able to replicate the error in python shell importing dateutil like we were and running a basic parse. So I fixed up the import and that should work now.

I also cleared up some of our dependency and requirement configurations. It makes setup.py a bit more complex as we try to determine whether or not to install the security extras for requests, but overall it should keep everything up-to-date.

The PR bumps pytx to version 0.5.1 and I'll be pushing that to PyPi shortly.

Thanks for the help! :)
